python/ray/_private/runtime_env/py_modules.py

```diff
@@ -139,9 +162,40 @@ async def create(
         # TODO(Catch-Bull): Refactor method create into an async process, and
         # make this method running in current loop.
         def _create():
-            module_dir = download_and_unpack_package(
-                uri, self._resources_dir, logger=logger
-            )
+            if is_whl_uri(uri):
+                # Download and install the wheel, and then delete the wheel file.
+                wheel_file = download_and_unpack_package(
+                    uri, self._resources_dir, logger=logger
+                )
+                module_dir = self._get_local_dir_from_uri(uri)
+
+                pip_install_cmd = [
+                    "pip",
+                    "install",
+                    wheel_file,
+                    f"--target={module_dir}",
+                ]
+                logger.info(
+                    "Running py_modules wheel install command: %s", str(pip_install_cmd)
+                )
+                exit_code, output = exec_cmd_stream_to_logger(pip_install_cmd, logger)
+
+                if Path(wheel_file).exists():
```

Comment:
can you move this code to a separate function instead of inlining all of the logic here? will help readability